Scripts used by Thom et al. 2023 GBE.

PipeMaster_Simulations: scripts used to simulate data under the coalescent and test alternative demographic models and estimate parameters with a neural network approach

diploS/HIC: Pipeline used to test for selective sweeps and hitchhiking using diploS/HIC 

1_GC_content.R: Script to calculate GC content for genomic windows

1_Running_twisst.txt: Commands used to calculate topology weight using twisst

1_SMC++_greg.job: Commands used to calculate the historical demography of Tapajos populations with SMC++

1_Species_tree_variables_relern_twist_sumStat_UCE_mtDNA_new.R: Script to obtain species trees for subsets of the genome based on quantiles (lower 10% and upper 90%) of summary statistics.

1_extracting_genetrees_with_variables_relern_twist_sumStat_UCE_mtDNA_new.R: Script to select gene trees based on quantiles (lower 10% and upper 90%) of summary statistics.

1_lostruct.job: Commands used to run local PCAs with lostruct

1_running_concat_tree.job: Commands to run a phylogenetic analysis with concatenated SNPs

